packages/back-end/src/routers/organizations/organizations.controller.ts

```diff
@@ -615,12 +619,16 @@ export async function getOrganization(req: AuthRequest, res: Response) {
   userInfo.forEach(({ id, email, verified, name, _id }) => {
     const memberInfo = members.find((m) => m.id === id);
     if (!memberInfo) return;
-    expandedMembers.push({
+
+    const userPermissions = getUserPermissions(id, org);
+
+    return expandedMembers.push({
       email,
       verified,
       name: name || "",
       ...memberInfo,
       dateCreated: memberInfo.dateCreated || _id.getTimestamp(),
+      userPermissions,
```

Comment:
@jdorn Given the current architecture, I think the current structure makes sense.

From my understanding, the `UserContext` makes two main calls - one to this endpoint (`getOrganization`) and another to the `getUser` endpoint, and it uses a combination of the data from both of these to build the `user`.

It feels a bit weird to make something user specific in the `getOrganization` method, and likewise, I don't think it makes sense to get the user's `userPermissions` within the `getUser` endpoint since a `user` record doesn't have a `role`.

If we wanted to make this a bit faster, and not return full permissions for every member in the org, it might make sense to build a new endpoint that takes in the user's id, and an org's id, and returns the permissions given those two factors. That would allow us to not have to needlessly build permissions for all users, without introducing any odd flows. This would probably just require some odd spreading within `UserContext`. 

We currently define the user via the organization.members array. If we go this route, we'd get nearly all the info we needed from organization.members, but then have to make that separate API call to get the `userPermissions`.

(If I misunderstood your comment, let me know, that's very possible)

Reply:
My concern was less about performance and more about privacy.  I don't know if we want a readonly user in an org to be able to get super detailed permission data about every other user.  I think they get back the role today, which also feels slightly invasive, but the full permission list feels worse.

I don't think it's too weird to have a user-specific field in the `getOrganization` response.  I'm also fine with a separate API call.

Reply:
@jdorn Ah - that makes sense. I've updated this so now we're only passing userPermissions for the currentUser.